providers/level.js

```diff
@@ -0,0 +1,140 @@
+const { sep, resolve } = require("path");
+const fs = require("fs-nextra");
+const level = require("level");
+const promisify = require("then-levelup");
+
+let baseDir;
+let loaded;
+
+/* eslint-disable no-confusing-arrow */
+exports.init = (client) => {
+  baseDir = resolve(client.clientBaseDir, "bwd", "provider", "level");
+  loaded = new client.methods.Collection();
+  fs.ensureDir(baseDir).catch(err => client.emit("log", err, "error"));
+  fs.readdir(baseDir)
+    .then(files => Promise.all(files.map(async (file) => {
+      const db = await promisify(level(baseDir + sep + file));
+      loaded.set(file, db);
+      return file;
+    })));
+};
+
+/* Table methods */
+
+/**
+ * Checks if a directory exists.
+ * @param {string} table The name of the table you want to check.
+ * @returns {Promise<boolean>}
+ */
+exports.hasTable = async table => loaded.has(table);
+
+/**
+ * Creates a new directory.
+ * @param {string} table The name for the new directory.
+ * @returns {Promise<Void>}
+ */
+exports.createTable = async (table) => {
+  const db = await promisify(level(baseDir + sep + table));
+  loaded.set(table, db);
+  return null;
+};
+
+/**
+ * Recursively deletes a directory.
+ * @param {string} table The directory's name to delete.
+ * @returns {Promise<Void>}
+ */
+exports.deleteTable = table => this.hasTable(table)
+  .then(exists => exists ? level.destroy(baseDir + sep + table).then(() => fs.remove(baseDir + sep + table)) : null);
+
+/* Document methods */
+
+/**
+ * Get all documents from a directory.
+ * @param {string} table The name of the directory to fetch from.
+ * @returns {Promise<Object[]>}
+ */
+exports.getAll = async (table) => {
+  if (!loaded.has(table)) return {};
```

Comment:
Fixed, also fixed the inserting that wasn't previously adding the ID to it. Or stringifying.